var/spack/repos/builtin/packages/survey/package.py

```diff
@@ -0,0 +1,209 @@
+        if spec.satisfies('+openmpi'):
+            mpi_options.append('-DOPENMPI_DIR=%s' % spec['openmpi'].prefix)
+        # mpich
+        if spec.satisfies('+mpich'):
+            mpi_options.append('-DMPICH_DIR=%s' % spec['mpich'].prefix)
+        # mpich2
+        if spec.satisfies('+mpich2'):
+            mpi_options.append('-DMPICH2_DIR=%s' % spec['mpich2'].prefix)
+        # mvapich
+        if spec.satisfies('+mvapich'):
+            mpi_options.append('-DMVAPICH_DIR=%s' % spec['mvapich'].prefix)
+        # mvapich2
+        if spec.satisfies('+mvapich2'):
+            mpi_options.append('-DMVAPICH2_DIR=%s' % spec['mvapich2'].prefix)
+        # mpt
+        if spec.satisfies('+mpt'):
+            mpi_options.append('-DMPT_DIR=%s' % spec['mpt'].prefix)
+
+        cmake_options.extend(mpi_options)
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Add in paths for finding package config files that tell us
+        # where to find these packages
+        cmake_args = [
+            '-DCMAKE_VERBOSE_MAKEFILE=ON',
+            '-DTLS_MODEL=implicit',
+            '-DLIBMONITOR_DIR=%s' % spec['libmonitor'].prefix,
+            '-DPAPI_DIR=%s' % spec['papi'].prefix,
+            '-DLIBIOMP_DIR=%s' % spec['llvm-openmp'].prefix,
+            '-DPYTHON_DIR=%s' % spec['python'].prefix,
+            '-DGOTCHA_DIR=%s' % spec['gotcha'].prefix
+        ]
+
+        # Add any MPI implementations coming from variant settings
+        self.set_mpi_cmake_options(spec, cmake_args)
+        return cmake_args
+
+    @property
+    def python_lib_dir(self):
+        python_vers_phrase = 'python{0}'.format(
+                             self.spec['python'].version.up_to(2))
+        return join_path('lib', python_vers_phrase)
+
+    @property
+    def site_packages_dir(self):
+        return join_path(self.python_lib_dir, 'site-packages')
+
+    def setup_run_environment(self, env):
+        """Set up the compile and runtime environments for a package."""
+
+        # Set SURVEY_MPI_IMPLEMENTATON to the appropriate mpi implementation
+        # This is needed by survey to deploy the correct
+        # mpi runtimes.
+        # Users may have to set the SURVEY_MPI_IMPLEMENTATION variable
+        # manually or create separate module files, if multiple mpi's
+        # are specified in the build
+        if self.spec.satisfies('+mpich'):
+            env.set('SURVEY_MPI_IMPLEMENTATION', "mpich")
+
+        if self.spec.satisfies('+mvapich'):
+            env.set('SURVEY_MPI_IMPLEMENTATION', "mvapich")
+
+        if self.spec.satisfies('+mvapich2'):
+            env.set('SURVEY_MPI_IMPLEMENTATION', "mvapich2")
+
+        if self.spec.satisfies('+mpt'):
+            env.set('SURVEY_MPI_IMPLEMENTATION', "mpt")
+
+        if self.spec.satisfies('+openmpi'):
+            env.set('SURVEY_MPI_IMPLEMENTATION', "openmpi")
+
+        env.prepend_path('MANPATH', self.spec.prefix.share.man)
+
+        env.prepend_path('PATH', self.spec['python'].prefix.bin)
```

Comment:
This is done by Spack, do you have a particular reason to add it?

Reply:
@haampie @adamjstewart These have been taken out with the 4 th commit.